File: bot/cogs/track.py
import discord
import requests
from discord import client
from discord.ext import commands
# from main import WELCOME_MESSAGE_CHANNEL_ID
from main import DOMAIN


class Track(commands.Cog):

    def __init__(self, client):
        self.client = client

    # Commands
    @ commands.command(brief="Tracks amazon links")
    async def track(self, context, prodLink, targetPrice):
        await context.send(f'Tracking price of {prodLink} for target price {targetPrice} and notifying on channel id {context.message.channel.id}')
        requests.post(f'{DOMAIN}/api/v1/amzn', {
            "url":
            prodLink,
            "targetPrice": targetPrice,
            "channelId": context.message.channel.id
        })


def setup(client):
    client.add_cog(Track(client))

File: bot/commands/track.js
const Discord = require("discord.js");
const axios = require("axios");
const { DOMAIN } = require("../config/config");

module.exports.run = async (client, message, args) => {
  // Call back response
  message.channel.send(
    `Tracking link ${args[0]} at target price of ${args[1]}`
  );

  // Post request to the api
  axios.post(`${DOMAIN}/api/v1/amzn`, {
    url: args[0],
    targetPrice: args[1],
    channelId: message.channel.id,
  });
  // .then((res) => console.log(res.data))
  // .catch((err) => console.log(err));
};

module.exports.help = {
  name: "track",
};
